# Renewable Energy Intelligence

## Data Understanding

### Load Dataset

In [2]:
import pandas as pd

df = pd.read_csv("../data/raw/Renewable.csv")

df.head()

,Time,Energy delta[Wh],GHI,temp,pressure,humidity,wind_speed,rain_1h,snow_1h,clouds_all,isSun,sunlightTime,dayLength,SunlightTime/daylength,weather_type,hour,month
0,2017-01-01 00:00:00,0,0.0,1.6,1021,100,4.9,0.0,0.0,100,0,0,450,0.0,4,0,1
1,2017-01-01 00:15:00,0,0.0,1.6,1021,100,4.9,0.0,0.0,100,0,0,450,0.0,4,0,1
2,2017-01-01 00:30:00,0,0.0,1.6,1021,100,4.9,0.0,0.0,100,0,0,450,0.0,4,0,1
3,2017-01-01 00:45:00,0,0.0,1.6,1021,100,4.9,0.0,0.0,100,0,0,450,0.0,4,0,1
4,2017-01-01 01:00:00,0,0.0,1.7,1020,100,5.2,0.0,0.0,100,0,0,450,0.0,4,1,1


### Dataset Shape

In [3]:
df.shape

(196776, 17)

### Data Types and Missing Values

In [4]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 196776 entries, 0 to 196775
Data columns (total 17 columns):
 #   Column                  Non-Null Count   Dtype  
---  ------                  --------------   -----  
 0   Time                    196776 non-null  str    
 1   Energy delta[Wh]        196776 non-null  int64  
 2   GHI                     196776 non-null  float64
 3   temp                    196776 non-null  float64
 4   pressure                196776 non-null  int64  
 5   humidity                196776 non-null  int64  
 6   wind_speed              196776 non-null  float64
 7   rain_1h                 196776 non-null  float64
 8   snow_1h                 196776 non-null  float64
 9   clouds_all              196776 non-null  int64  
 10  isSun                   196776 non-null  int64  
 11  sunlightTime            196776 non-null  int64  
 12  dayLength               196776 non-null  int64  
 13  SunlightTime/daylength  196776 non-null  float64
 14  weather_type            196776 

### Target Variable Summary

In [5]:
df["Energy delta[Wh]"].describe()

count    196776.000000
mean        573.008228
std        1044.824047
min           0.000000
25%           0.000000
50%           0.000000
75%         577.000000
max        5020.000000
Name: Energy delta[Wh], dtype: float64

### Zero Production Analysis

In [6]:
(df["Energy delta[Wh]"] == 0).sum()

np.int64(100850)

### Relationship Between Sunlight and Zero Production

In [8]:
df[df["Energy delta[Wh]"] == 0]["isSun"].value_counts()

isSun
0    94460
1     6390
Name: count, dtype: int64

**Observation:**
94,460 zero-production observations occur when isSun = 0.

**Interpretation:**
This suggests the dataset behaves consistently with solar generation, as energy production is generally absent when sunlight is unavailable.